src/plugins/gmx/gmxplugin.cpp

```diff
@@ -110,16 +110,69 @@ bool GmxPlugin::write(const Map *map, const QString &fileName)
     writeProperty(stream, map, "speed", 30);
     writeProperty(stream, map, "persistent", false);
     writeProperty(stream, map, "clearDisplayBuffer", true);
+    writeProperty(stream, map, "clearViewBackground", false);
+    writeProperty(stream, map, "enableViews", false);
 
     stream.writeTextElement("isometric", toString(map->orientation() == Map::Orientation::Isometric));
 
+    // Write out views
+    // Last view in Object layer is the first view in the room
+    stream.writeStartElement("views");
+    LayerIterator iterator(map);
+    int viewCount = 0;
+    while (const Layer *layer = iterator.next()) {
+
+        if (layer->layerType() != Layer::ObjectGroupType)
+            continue;
+
+        const ObjectGroup *objectLayer = static_cast<const ObjectGroup*>(layer);
+
+          for (const MapObject *object : objectLayer->objects()) {
+            const QString type = effectiveObjectType(object);
+            if (type.isEmpty())
```

Comment:
An additional empty check isn't needed, cause it is already covered by the next check.